code/game/objects/items/blueprints.dm

```diff
@@ -216,6 +218,7 @@
 		A.setup(str)
 		A.contents += turfs
 		A.SetDynamicLighting()
+	A.has_gravity = old.has_gravity
```

Comment:
Sans the gravity generator, your code has the exact same function as has_gravity()